packages/assistify-ai/server/methods/SmartiReload.js

```diff
@@ -0,0 +1,154 @@
+/* globals SystemLogger, RocketChat */
+
+import {SmartiProxy, verbs} from '../SmartiProxy';
+
+Meteor.methods({
+	triggerFullResync() {
+
+		const query = {$or: [{outOfSync: true}, {outOfSync: {$exists: false}}]};
+
+		query.t = 'r';
+		const requests = RocketChat.models.Rooms.model.find(query).fetch();
+		SystemLogger.debug('Number of Requests to sync: ', requests.length);
+		for (let i=0; i < requests.length; i++) {
+			Meteor.defer(()=>Meteor.call('tryResync', requests[i]._id));
+		}
+
+		query.t = 'e';
+		const topics = RocketChat.models.Rooms.model.find(query).fetch();
+		SystemLogger.debug('Number of Topics to sync: ', topics.length);
+		for (let i=0; i < topics.length; i++) {
+			Meteor.defer(()=>Meteor.call('tryResync', topics[i]._id));
+		}
+
+		return {
+			message: 'sync-triggered-successfully'
+		};
+	}
+});
+
+Meteor.methods({
+	markMessageAsSynced(messageId) {
+		const messageDB = RocketChat.models.Messages;
+		const message = messageDB.findOneById(messageId);
+		const lastUpdate = message ? message._updatedAt : 0;
+		if (lastUpdate) {
+			messageDB.model.update(
+				{_id: messageId},
+				{
+					$set: {
+						lastSync: lastUpdate
+					}
+				});
+			SystemLogger.debug('Message Id: ', messageId, ' has been synced');
+		} else {
+			SystemLogger.debug('Message Id: ', messageId, ' can not be synced');
+		}
+	}
+});
+
+Meteor.methods({
+	markRoomAsUnsynced(rid) {
+		RocketChat.models.Rooms.model.update(
+			{_id: rid},
+			{
+				$set: {
+					outOfSync: true
+				}
+			});
+		SystemLogger.debug('Room Id: ', rid, ' is out of sync');
+	}
+});
+
+Meteor.methods({
+	tryResync(rid) {
+		SystemLogger.debug('Sync all unsynced messages in room: ', rid);
+		const room = RocketChat.models.Rooms.findOneById(rid);
+		const messageDB = RocketChat.models.Messages;
+		const unsync = messageDB.find({ lastSync: { $exists: false }, rid, t: { $exists: false } }).fetch();
+		if (unsync.length === 0) {
+			SystemLogger.debug('Room is already in sync');
+			return true;
+		} else {
+			SystemLogger.debug('Messages out of sync: ', unsync.length);
+		}
+		let conversation;
+
+		// conversation exists for channel?
+		SystemLogger.debug('Smarti - Trying legacy service to retrieve conversation ID...');
```

Comment:
we have this code a couple of times. I extracted this in https://github.com/assistify/Rocket.Chat/blob/fix/smarti/message-lifecycle/packages/assistify-ai/server/lib/SmartiAdapter.js#L42-L66

I believe we should not merge this PR before the message lifecycle - or better merge `fix/smarti/message-lifecycle` into this branch.